src/server/api/routers/event.ts

```diff
@@ -13,6 +15,22 @@ const byDateRangeSchema = z.object({
   startTime: z.date(),
   endTime: z.date(),
 });
+const filterableSchema = z.object({
+  startTime: z.union([
+    z.literal('now'),
+    z.object({ days: z.number() }),
+    z.custom<DateRange>((val) => isDateRange(val)),
+  ]),
+  order: z.union([
+    z.literal('soon'),
+    z.literal('later'),
+    z.literal('shortest duration'),
+    z.literal('longest duration'),
```

Comment:
Is there a reason why you're using a union of literals over a zod enum?

Reply:
changed it to an enum